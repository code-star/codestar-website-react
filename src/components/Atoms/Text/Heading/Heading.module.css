.headingWhite {
	color: #fff;
}

.headingBlack {
	color: #000;
}
