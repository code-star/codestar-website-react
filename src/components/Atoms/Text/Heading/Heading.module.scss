@import '../../../../theming/config.module.scss';

.heading {
  &White {
    color: $color-white;
  }

  &Black {
    color: $color-black;
  }
}
