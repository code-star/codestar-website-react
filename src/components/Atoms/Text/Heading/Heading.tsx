import React from 'react';
import classNames from 'classnames/bind';
import style from './Heading.module.css';

const cx = classNames.bind(style);

interface IHeadingProps {
  type: string;
  color: string;
  text: string;
  className?: string;
}

const Heading = ({
  type,
  color = 'black',
  text,
  className = '',
}: IHeadingProps) => {
  const classNamesFromProps = {
    headingWhite: color === 'white',
    headingBlack: color === 'black',
  };

  if (type === 'h1') {
    return <h1 className={cx(className, classNamesFromProps)}>{text}</h1>;
  } else if (type === 'h2') {
    return <h2 className={cx(className, classNamesFromProps)}>{text}</h2>;
  } else if (type === 'h3') {
    return <h3 className={cx(className, classNamesFromProps)}>{text}</h3>;
  } else if (type === 'h4') {
    return <h4 className={cx(className, classNamesFromProps)}>{text}</h4>;
  }
  return null;
};

export default Heading;
