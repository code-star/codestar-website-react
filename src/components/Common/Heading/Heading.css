.heading--white {
	color: #fff;
}

.heading--black {
	color: #000;
}
