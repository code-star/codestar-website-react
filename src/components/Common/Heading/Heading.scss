@import '../../../theming/config.scss';

.heading {
	&White {
		color: $color-white;
	}

	&Black {
		color: $color-black;
	}
}
