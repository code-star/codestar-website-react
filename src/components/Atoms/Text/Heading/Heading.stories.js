import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs';
import Heading from './Heading.tsx';

storiesOf('Components/Atoms/Text', module)
  .addDecorator(withKnobs)
  .add('Heading', () => {
    const type = select('Type', ['h1', 'h2', 'h3', 'h4'], 'h2');

    return <Heading type={type} text={`Awesome ${type} heading`} />;
  });
