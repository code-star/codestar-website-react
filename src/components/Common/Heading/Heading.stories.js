import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select, text } from '@storybook/addon-knobs';
import Heading from './Heading.tsx';

storiesOf('Components', module)
	.addDecorator(withKnobs)
	.add('Heading', () => {
		const type = select('Type', ['h1', 'h2', 'h3', 'h4'], 'h2');

		return <Heading type={type} text="Some Heading" />;
	});
