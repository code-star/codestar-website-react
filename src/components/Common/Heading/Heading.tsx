import React, { SFC } from 'react';
import cx from 'classnames';
import './Heading.css';

interface IHeadingProps {
	type: string;
	color: string;
	text: string;
}

const Heading: SFC<IHeadingProps> = ({ type, color = 'black', text }) => {
	if (type === 'h1') {
		return (
			<h1
				className={cx('heading', {
					'heading--white': color === 'white',
					'heading--black': color === 'black',
				})}
			>
				{text}
			</h1>
		);
	} else if (type === 'h2') {
		return (
			<h2
				className={cx('heading', {
					'heading--white': color === 'white',
					'heading--black': color === 'black',
				})}
			>
				{text}
			</h2>
		);
	} else if (type === 'h3') {
		return (
			<h3
				className={cx('heading', {
					'heading--white': color === 'white',
					'heading--black': color === 'black',
				})}
			>
				{text}
			</h3>
		);
	} else if (type === 'h4') {
		return (
			<h4
				className={cx('heading', {
					'heading--white': color === 'white',
					'heading--black': color === 'black',
				})}
			>
				{text}
			</h4>
		);
	}
	return null;
};

export default Heading;
