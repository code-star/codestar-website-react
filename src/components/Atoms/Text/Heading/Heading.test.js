import React from 'react';
import { shallow } from 'enzyme';

import Heading from './Heading';

const getComponent = type => (
	<Heading type={type} text={`Awesome ${type} heading`} />
);

const renderShallow = () => {
	return shallow(getComponent('h1'));
};

describe('<Heading />', () => {
	let wrapper;

	describe('Instance', () => {
		test('must be an instance of Heading', () => {
			wrapper = renderShallow();
			expect(wrapper.find('Heading')).toBeTruthy();
		});
	});

	describe('Snaphot', () => {
		test('must match h1', () => {
			expect(global.renderToJSON(getComponent('h1'))).toMatchSnapshot();
		});

		test('must match h2', () => {
			expect(global.renderToJSON(getComponent('h2'))).toMatchSnapshot();
		});

		test('must match h3', () => {
			expect(global.renderToJSON(getComponent('h3'))).toMatchSnapshot();
		});

		test('must match h4', () => {
			expect(global.renderToJSON(getComponent('h4'))).toMatchSnapshot();
		});
	});
});
