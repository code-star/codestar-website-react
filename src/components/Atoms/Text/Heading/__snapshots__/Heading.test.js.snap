// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> Snaphot must match h1 1`] = `
<h1
  className="headingBlack"
>
  Awesome h1 heading
</h1>
`;

exports[`<Heading /> Snaphot must match h2 1`] = `
<h2
  className="headingBlack"
>
  Awesome h2 heading
</h2>
`;

exports[`<Heading /> Snaphot must match h3 1`] = `
<h3
  className="headingBlack"
>
  Awesome h3 heading
</h3>
`;

exports[`<Heading /> Snaphot must match h4 1`] = `
<h4
  className="headingBlack"
>
  Awesome h4 heading
</h4>
`;
